### DISCUSSION TITLE
changing model on the site device after moving currect device

### DISCUSSION THREAD

**Discussion by Dees7** [netbox-community/netbox#8324] [Q&A] [answered]
Hello.
Is it possible to simple changing device model and copy all connected links (might be copy descriptions too), some virtual Interfaces.
I'd not like editing current cause this device was moved to other Site and it has it's Asset Tag\Serial Number.

**Comment by candlerb** [comment, marked as answer]
It sounds like you've replaced device A with a different device B (which is a different type), but have plugged all the connections which were originally in A to B.

Netbox isn't an asset management system, so normally when swapping out a unit, you'd simply change the asset tag / serial number of the "device" in the database, and leave all the connections alone.  An asset management system would track the lifecycle of the asset from purchase to disposal, being put into a spares pool, deployed, returned to manufacturer for repair etc; Netbox doesn't do that.

If you think of a "device" in Netbox as a record of "where something is deployed right now", or "what's sitting in this rack position right now", and that thing can change, then maybe it makes more sense.

You've also changed the device type.  Fortunately, you _are_ allowed to change the device type in Netbox; this does not affect anything else on the device.  In particular, it does not change the interfaces or ports on the device (these are only copied from the device type when the device is first created), hence it does not change any connections.  However you may need to go through all the interfaces and rename them to match the new device, and/or add extra interfaces or remove ones which shouldn't be there.  [This report](https://github.com/netbox-community/reports/blob/master/reports/dcim-reports/missing_device_type_components.py) may be helpful.

If you still want to go the other way, and export and reimport all virtual interfaces and all cables, that's not straightforward.  What makes it tricky is that the CSV export and CSV import formats in Netbox are not compatible with each other.  I've made it work for cables by writing an export template - you can find it [here](https://github.com/netbox-community/netbox/discussions/7792#discussioncomment-1631444).  This exports cables in a form which is compatible with CSV import.  You'll still have to delete all the old cables, and possibly modify the CSV data to update device name and/or interface names to match the new device, before importing.

**Reply by Dees7** [reply]
I see. Thank you very much. 
The main reason keeping device "linked" to real one is History. I'd like to see all changes with real device (through NB history). After I move device on other site it the same device I'd like to know all about it.
